C-Sharp/VectorDistanceCalculation/PE11VectorDistanceCalculation/Menu.cs

```diff
@@ -37,29 +45,33 @@ public static void Run()
 
         }
 
-        private static int getValidUserInput()
+        private static int getValidUserInput()//Start ALL methods with Capital letter
         {
             displaySelectionMenu();
-            string userInput = Console.ReadLine();
+            string userInput = Console.ReadLine();//var
             try
             {
                 int selection = int.Parse(userInput);
-                if (!validSelections.Any(s => s == selection)) throw new Exception();
+                //why have valid selections? why not return invalid error or something? you have a switch statement
+                if (!validSelections.Any(s => s == selection)) throw new Exception();//use {}
                 return selection;
 
             }
-            catch (Exception)
+            catch (Exception) // dont catch exception, if you dont know what exception here
+            //see program.cs
             {
                 Console.WriteLine("You have entered an invalid selection.  Please try again.\n\n");
-                return getValidUserInput();
+                return getValidUserInput();//oh boy this can cause some issues
```

Comment:
Is it the use of the recursive method itself or the placement of it?  Do you prefer using while loops or another?

Reply:
The problem is there is no exit.

Reply:
I see what you are saying...  My intent is to now allow invalid input, but if the user continues to make an invalid input, it will have no way to exit the loop, and may even cause stackoverflow.  I need to think about how to terminate it, maybe by adding a counter to keep track of attempts before exiting out...